### PR TITLE
[stable/ghost] Add JSON Schema to Ghost chart

### PR DESCRIPTION
**Pull request by andresmgot**
#### What this PR does / why we need it:

With Helm v3, it's now possible to include a [JSON Schema](https://json-schema.org/) in the chart package that validates that the given values satisfy the expected schema. This PR includes a basic JSON schema for the Ghost chart. It doesn't cover all the possible values of the chart but just the ones that, in my opinion, are more likely to be modified.

Apart from validating that the values complies with the schema, the next version of Kubeapps will use these schemas to render a simple form so the chart containing it is easier to configure and deploy. See https://github.com/kubeapps/kubeapps/blob/master/docs/developer/basic-form-support.md for more information.

Please let me know if you think any parameter should be included/removed from this list. This is an example of the file rendered by Kubeapps:

Screenshot from 2019-10-23 10-25-06

cc/ @absoludity

Caveat: I was not able to add the hostname to the form because it's current format (an array of objects). Since the standard way of setting these parameters is not clear I will leave that out of the form for the moment.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by absoludity** [comment]
Thanks Andres.

> Caveat: I was not able to add the hostname to the form because it's current format (an array of objects). Since the standard way of setting these parameters is not clear I will leave that out of the form for the moment.

Why do you think that the current format of `ghostHost` is an array of objects? I understood it is just a string (the hostname) as per https://github.com/helm/charts/tree/master/stable/ghost#configuration (I'm not sure if you're confusing the ingress.hosts which is an array of objects, or if I'm incorrectly not taking the ingress into account).

The install isn't complete with `ghostHost`: https://github.com/helm/charts/blob/master/stable/ghost/templates/NOTES.txt#L1

**Review by alemorcuq — APPROVED** [review]
Looks good to me.

**Comment by alemorcuq** [comment]
> Thanks Andres.
> 
> > Caveat: I was not able to add the hostname to the form because it's current format (an array of objects). Since the standard way of setting these parameters is not clear I will leave that out of the form for the moment.
> 
> Why do you think that the current format of `ghostHost` is an array of objects? I understood it is just a string (the hostname) as per https://github.com/helm/charts/tree/master/stable/ghost#configuration (I'm not sure if you're confusing the ingress.hosts which is an array of objects, or if I'm incorrectly not taking the ingress into account).
> 
> The install isn't complete with `ghostHost`: https://github.com/helm/charts/blob/master/stable/ghost/templates/NOTES.txt#L1

I believe Andres is talking about the Ingress configuration, for which the `hosts` are indeed an array of objects: https://github.com/helm/charts/blob/66cb3a49f64dae71a27faf4f3b41db40c69bcff8/stable/ghost/values.yaml#L236

**Comment by absoludity** [comment]
> > Thanks Andres.
> > > Caveat: I was not able to add the hostname to the form because it's current format (an array of objects). Since the standard way of setting these parameters is not clear I will leave that out of the form for the moment.
> > 
> > 
> > Why do you think that the current format of `ghostHost` is an array of objects? I understood it is just a string (the hostname) as per https://github.com/helm/charts/tree/master/stable/ghost#configuration (I'm not sure if you're confusing the ingress.hosts which is an array of objects, or if I'm incorrectly not taking the ingress into account).
> > The install isn't complete with `ghostHost`: https://github.com/helm/charts/blob/master/stable/ghost/templates/NOTES.txt#L1
> 
> I believe Andres is talking about the Ingress configuration, for which the `hosts` are indeed an array of objects:
> 
> https://github.com/helm/charts/blob/66cb3a49f64dae71a27faf4f3b41db40c69bcff8/stable/ghost/values.yaml#L236

Yes, I think so too. Sorry Ale, there was extra context in a private slack conversation where I'd asked if we could have `ghostHost` as one of the options on the simple form (since it is required for the chart to deploy correctly). That's why I was wondering if Andres was looking at the ingress config instead of `ghostHost`. I don't mind the ingress being in the advance form / values.yaml, but I think it's important that `ghostHost` is exposed in the simple form. Hope that clarifies :)

**Comment by andresmgot** [comment]
Correct, I was talking about the ingress setup, since it's different from the one for wordpress. 

In any case, I missed that `ghostHost` is necessary (it doesn't have any comment in the values.yaml). I think that's an usability issue of the chart. When using `ghostHost` to set the hostname, there is nothing that aligns that with the ingress setup.

File: stable/ghost/values.yaml [hunk 1]
```diff
@@ -98,20 +98,21 @@ allowEmptyPassword: "yes"
 ## External database configuration
 ##
 externalDatabase:
+  ## All of these values are only used when mariadb.enabled is set to false
   ## Database host
-  # host:
+  host: localhost
```
Comment (carrodher) [inline stable/ghost/values.yaml:103]: Can you edit this new default value in the README table listing the parameters?

File: stable/ghost/values.yaml [hunk 2]
```diff
@@ -98,20 +98,21 @@ allowEmptyPassword: "yes"
...
   ## Database password
-  # password:
+  password: ""
 
   ## Database name
-  # database: bitnami_ghost
+  database: bitnami_ghost
+
+  ## Database port number
+  port: 3306
```
Comment (carrodher) [inline stable/ghost/values.yaml:115]: Same

File: stable/ghost/values.yaml [hunk 3]
```diff
@@ -98,20 +98,21 @@ allowEmptyPassword: "yes"
...
-  # port:
-
-  ## Database user
-  # user: bn_ghost
+  ## non-root Username for Wordpress Database
+  user: bn_ghost
 
   ## Database password
-  # password:
+  password: ""
```
Comment (carrodher) [inline stable/ghost/values.yaml:109]: Same. `nil` vs `""`

**Comment by carrodher** [comment]
Can you take a look at the inline comments? In the same way, you also need to bump again the chart version because the current one is `version: 7.2.12`.

**Pull request merged**